playbooks/roles/logstash/templates/logstash.conf.j2

```diff
@@ -1,43 +1,115 @@
+    }
+  } else if ("supervisor_error" in [openedx_log_format]) {
+    grok {
+      match => { "message" => "%{TIMESTAMP_ISO8601:syslog_timestamp} %{WORD:log_level} %{GREEDYDATA:syslog_message}" }
+      match => { "message" => "\[%{TIMESTAMP_ISO8601:syslog_timestamp}\: %{LOGLEVEL:log_level}\/%{USERNAME:supervisor_program}\] %{GREEDYDATA:syslog_message}" }
+      tag_on_failure => []
+    }
+  } else if ("supervisor_stdout" in [openedx_log_format]) { 
+    grok {
+      match => { "message" => "%{GREEDYDATA:syslog_message}" }
+    }
+  } else if ("supervisor_log" in [openedx_log_format]) {
+    grok {
+      match => { "message" => "%{TIMESTAMP_ISO8601:syslog_timestamp} %{LOGLEVEL:log_level} %{WORD:supervisor_result}\: %{GREEDYDATA:syslog_message}" }
+    }
+  } else if ("supervisor_error" in [openedx_log_format] and "xqueue" in [service_variant]) {
+    grok {
+      match => { "message" => "%{TIMESTAMP_ISO8601:syslog_timestamp} \[%{POSINT:syslog_pid}\] \[%{LOGLEVEL:log_level}\] %{GREEDYDATA:syslog_message}" }
+    }
+  } else if ("supervisor_stdout" in [openedx_log_format]) {
+    grok {
+      match => { "message" => "%{TIMESTAMP_ISO8601:syslog_timestamp} %{LOGLEVEL:log_level} %{POSINT:syslog_pid} \[%{DATA:queue_service}\] %{DATA:filename}\:%{INT:line_number} \- %{GREEDYDATA:syslog_message}" }
+    }
+  } else if ("rabbitmq" in [openedx_log_format]) {
+    grok {
+      match => { "message" => "%{WORD}\=%{QUOTEDSTRING:syslog_timestamp} %{WORD}\=%{QUOTEDSTRING:rabbitmq_vhost} %{WORD}\=%{QUOTEDSTRING:rabbitmq_queue} %{WORD}\=%{INT:rabbitmq_length}" }
+    }
+  } else if ("mongo" in [openedx_log_format]) {
+    grok {
+      match => { "message" => "%{TIMESTAMP_ISO8601:syslog_timestamp} ?\[?%{DATA:mongo_context}?\]? %{GREEDYDATA:syslog_message}" }
+    } 
+  } else if ("tracking" in [opendedx_log_format]) { 
+    grok {
+      match => { "message" => "%{GREEDYDATA:syslog_message}" }
+    }
+  } else if ("logstash" in [openedx_log_format]) {
+    grok {
+      match => { "message" => "\{\:%{WORD}\=\>\"%{TIMESTAMP_ISO8601:syslog_timestamp}\"\, %{GREEDYDATA:syslog_message}\}" }
+    }
+  } else if ("elasticsearch" in [openedx_log_format]) {
+    grok {
+      match => { "message" => "(?:\[%{TIMESTAMP_ISO8601:syslog_timestamp}\]\[%{WORD:log_level}? \])%{GREEDYDATA:syslog_message}" }
+    }
+  } else if ("rsyslog" in [openedx_log_format]) {
+    grok {
+      match => { "message" => "%{SYSLOGTIMESTAMP:syslog_timestamp} %{WORD:syslog_hostname} %{DATA:syslog_program}(?:\[%{POSINT:syslog_pid}\])?: %{GREEDYDATA:syslog_message}" }
+      match => { "message" => "%{SYSLOGTIMESTAMP:syslog_timestamp} %{WORD:syslog_hostname} \[%{WORD}\=%{DATA:service_variant}\]\[%{DATA:service_program}\]\[%{WORD}\:%{USERNAME:env}\] %{WORD:log_level} \[%{WORD} ? %{POSINT:syslog_pid}\] \[%{DATA:filename}\:%{INT:line_number}\] \- %{GREEDYDATA:syslog_message}" }
+      tag_on_failure => []
+    }
+  }
```

Comment:
flag topbeat, if at all possible.

add an else block to catch-all; will still need handled via filebeat, but just in case..